basestar-storage-sql/src/main/java/io/basestar/storage/sql/SQLStorage.java

```diff
@@ -657,6 +658,24 @@ public StorageTraits storageTraits(final ReferableSchema schema) {
         return Collections.emptySet();
     }
 
+    @Override
+    public CompletableFuture<Long> increment(final SequenceSchema schema) {
+
+        return withContext(context -> {
+
+            final org.jooq.Name sequenceName = strategy.getNamingStrategy().sequenceName(schema);
+            return strategy.dialect().incrementSequence(context, sequenceName).fetchAsync().thenApply(result -> {
+
+                final Iterator<Record1<Long>> iterator = result.iterator();
+                if (iterator.hasNext()) {
+                    return iterator.next().value1();
+                } else {
+                    throw new IllegalStateException("Sequence yielded no value");
```

Comment:
add the sequence name to the error?

Reply:
This is a very low-level error tbh, it would only be thrown if the creator of a dialect failed to implement the interface method/failed to throw a better error, is not something a user should be able to trigger with configuration